examples/notebooks/beam-ml/mltransform/compute_and_apply_vocab.ipynb

```diff
@@ -0,0 +1,326 @@
+            "Row(feature=array([-1, -1, -1,  2,  0, -1]))\n",
+            "Row(feature=array([0, 1, 4, 3]))\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## `MLTransform` for inference workloads\n",
+        "\n",
+        "When `MLTransform` is in `write` mode, it produces artifacts, such as vocabulary files for `ComputeAndApplyVocabulary`. When `MLTransform` is used `read` mode, it uses the previously generated vocabulary files to map the incoming text data. If the incoming vocabulary isn't found in the generated vocabulary, then the incoming vocabulary is mapped to a `default_value` provided during the `write` mode. In this case, the `default_value` is `-1`."
+      ],
+      "metadata": {
+        "id": "PEqYpCexybO4"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "test_documents = [\n",
+        "    {'feature': 'wizards are flying in the sky'},\n",
+        "    {'feature': 'I love dogs'}\n",
+        "]\n",
+        "\n",
+        "with beam.Pipeline() as pipeline:\n",
+        "  data_pcoll = pipeline | \"CreateData\" >> beam.Create(test_documents)\n",
+        "  # Compute and apply vocabulary by using MLTransform.\n",
+        "  transformed_pcoll = (\n",
+        "      data_pcoll\n",
+        "      | \"MLTransform\" >> MLTransform(read_artifact_location=artifact_location))\n",
+        "\n",
+        "  transformed_pcoll | \"Print\" >> beam.Map(print)"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "x-LRXUaTyX_B",
+        "outputId": "d7946bdc-8c1f-49cc-a7f9-872d85e347dc"
+      },
+      "execution_count": 21,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "WARNING:absl:You are passing instance dicts and DatasetMetadata to TFT which will not provide optimal performance. Consider following the TFT guide to upgrade to the TFXIO format (Apache Arrow RecordBatch).\n",
+            "WARNING:absl:You are outputting instance dicts from `TransformDataset` which will not provide optimal performance. Consider setting  `output_record_batches=True` to upgrade to the TFXIO format (Apache Arrow RecordBatch). Encoding functionality in this module works with both formats.\n"
+          ]
+        },
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Row(feature=array([ 5, 18, -1,  2,  0,  6]))\n",
+            "Row(feature=array([-1, -1, 15]))\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "When you specify `read_artifact_location`, the previous step doesn't pass transforms to `MLTransform`. Instead, `MLTransform` saves the artifacts and the transforms produced in the location specified by `write_artifact_location`."
```

Comment:
I found this wording a little unclear, maybe this would help:

```suggestion
        "When you specify `read_artifact_location`, you don't have to pass any transforms to `MLTransform`. Instead, `MLTransform` saves the artifacts and the transforms produced in the location specified by `write_artifact_location`."
```